From 60c64852962d5cd415ec4eda023c9b1437d39691 Mon Sep 17 00:00:00 2001
From: Ondrej Jirman <megous@megous.com>
Date: Sun, 1 Sep 2019 21:23:57 +0200
Subject: [PATCH 15/26] thermal: sun8i: Add some comments about H3

H3 is a mess. There are multiple different equations floating
around in various BSPs and the user manual.

Signed-off-by: Ondrej Jirman <megous@megous.com>
---
 drivers/thermal/sun8i_thermal.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/drivers/thermal/sun8i_thermal.c b/drivers/thermal/sun8i_thermal.c
index 0c186596d5e8..20f2bfeef4cb 100644
--- a/drivers/thermal/sun8i_thermal.c
+++ b/drivers/thermal/sun8i_thermal.c
@@ -564,6 +564,15 @@ static const struct ths_thermal_chip sun8i_h3_ths = {
 	// original BSP divs by 8.253 adds 217
 	.scale = 1211,
 	.offset = 217000,
+
+	// datasheet says T = (Tem - 2794)/-14.882
+	//.scale = 672,
+	//.offset = 187743,
+
+	// 4.9 BSP says: T = -0.118 * Tem + 256 = 256 - Tem/8.4746
+	//.scale = 1180,
+	//.offset = 256000 + 15000,
+
 	.temp_data_base = SUN8I_THS_TEMP_DATA,
 	.calibrate = sun8i_h3_ths_calibrate,
 	.init = sun8i_h3_thermal_init,
-- 
2.23.0

